libs/api/protocol/src/lib/core/query/get-protocol-entries.query.ts

```diff
@@ -0,0 +1,70 @@
+import { Inject } from '@nestjs/common';
+import { IQueryHandler, QueryHandler } from '@nestjs/cqrs';
+
+import { Page } from '../entity/page.entity';
+import { ProtocolEntryBase } from '../entity/protocol-entries/protocol-entry-base.entity';
+import {
+	PROTOCOL_ENTRY_REPOSITORY,
+	ProtocolEntryRepository,
+} from '../repository/protocol-entry.repository';
+
+export class GetProtocolEntriesQuery {
+	constructor(
+		public readonly organizationId: string,
+		public readonly count: number,
+		public readonly sort: 'asc' | 'desc',
+		public readonly startingFrom?: Date,
+	) {}
+}
+
+@QueryHandler(GetProtocolEntriesQuery)
+export class GetProtocolEntriesHandler
+	implements IQueryHandler<GetProtocolEntriesQuery>
+{
+	constructor(
+		@Inject(PROTOCOL_ENTRY_REPOSITORY)
+		private readonly repository: ProtocolEntryRepository,
+	) {}
+
+	async execute({
+		organizationId,
+		count,
+		sort,
+		startingFrom,
+	}: GetProtocolEntriesQuery): Promise<Page<ProtocolEntryBase>> {
```

Comment:
At first, I thought so as well. However, as we fetch the list of documents first and only then the info about previous/next **based on the first/last documents of the list**, we will always get a valid state. Even more, the state of `hasPrevious`/`hasNext` will be "more up-to-date" by not using a TA because they take into account TAs which happened between fetching the list of documents and fetching information for the `hasPrevious`/`hasNext` flags.

Only the total count can be influenced negatively by TAs applied to the collection between the different requests. However, as we are talking about a difference by <10 documents, and in most cases none at all, while having a rather large total (I estimate four to five digit numbers rather quickly due to the automated status entries), and we don't rely on the precision of the total count, I find it unnecessary, even damaging, to put the queries into a TA and thus put more load on to the database.